lua/cmp_tabnine/source.lua

```diff
@@ -286,16 +286,19 @@ Source._on_stdout = function(_, data, _)
               old_suffix = string.sub(old_suffix, 1, -2)
             end
 
+            local range = {
+              start = { line = cursor.line, character = cursor.col - #old_prefix - 1 },
+              ['end'] = { line = cursor.line, character = cursor.col + #old_suffix - 1 },
+            }
+
             local item = {
               label = newText,
               filterText = newText,
               data = result,
               textEdit = {
-                range = {
-                  start = { line = cursor.line, character = cursor.col - #old_prefix - 1 },
-                  ['end'] = { line = cursor.line, character = cursor.col + #old_suffix - 1 },
-                },
                 newText = newText,
+                insert = range, -- May be better to exclude the trailing part of old_suffix since it's 'replaced'?
```

Comment:
Why specify both replace and insert? shouldnt we be supplying only one?

Reply:
Range in `insert` field is used in insert behavior: https://github.com/hrsh7th/nvim-cmp/blob/1797f9e1acd2e0b5b4b6805928aebc1dcc0ecbff/lua/cmp/entry.lua#L298-L305, I think it should work as same for both modes.